src/visGeometry/index.ts

```diff
@@ -845,9 +845,12 @@ class VisGeometry {
         if (parent === undefined || parent == null) {
             return;
         }
-
+        if (parent["data-has-simularium-viewer-canvas"]) {
+            return;
+        }
         this.threejsrenderer = this.createWebGL();
         parent.appendChild(this.threejsrenderer.domElement);
+        parent["data-has-simularium-viewer-canvas"] = true;
```

Comment:
I like this, this is a nice fix.  I still kinda think it could do more intelligent checking (parent!==canvas.parent) and other stuff but it gets more complicated quickly.